include/keisan/matrix/matrix.hpp

```diff
@@ -74,10 +75,30 @@ class Matrix
   double * operator[](size_t pos);
   const double * operator[](size_t pos) const;
 
+  bool inverse();
+
 private:
   double data[M * N];
 };
 
+struct Point3;
```

Comment:
i suggest to put forward declaration (`struct x;`) on top most of this file, right after include declarations (`#include <x>`).